test/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -15,17 +15,47 @@ package org.codice.compliance.verification.core
 
 import org.apache.cxf.rs.security.saml.sso.SSOConstants.SIGNATURE
 import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLSpecRefMessage
 import org.codice.compliance.SAMLSpecRefMessage.SAMLCore_5_4_1
 import org.codice.compliance.SAMLSpecRefMessage.SAMLCore_5_4_2_a
 import org.codice.compliance.SAMLSpecRefMessage.SAMLCore_5_4_2_b
 import org.codice.compliance.SAMLSpecRefMessage.SAMLCore_5_4_2_b1
 import org.codice.compliance.SAMLSpecRefMessage.SAMLCore_6_1_b
 import org.codice.compliance.allChildren
 import org.codice.compliance.children
+import org.codice.compliance.utils.TestCommon
 import org.codice.compliance.utils.TestCommon.Companion.ELEMENT
 import org.w3c.dom.Node
 
 class CoreVerifier(val node: Node) {
+
+    /**
+     * Verifies that the response has an appropriate error status code.
+     * This should be called explicitly if an error is expected
+     */
+    fun verifyErrorStatusCode(samlErrorCode: SAMLSpecRefMessage) {
+        val statusCodes = node.allChildren("StatusCode")
+        if (statusCodes.isEmpty())
+            throw SAMLComplianceException.createWithPropertyMessage(samlErrorCode,
+                    property = "Status Codes",
+                    actual = "[]",
+                    expected = "<StatusCode Value=...>")
+
+        statusCodes.forEach {
+            val status = it.attributes.getNamedItem("Value").textContent
+            if (status == TestCommon.SUCCESS
+                    && !TestCommon.TOP_LEVEL_STATUS_CODES.contains(status)
+                    && !TestCommon.SECOND_LEVEL_STATUS_CODES.contains(status)) {
+                throw SAMLComplianceException.createWithPropertyMessage(samlErrorCode,
+                        property = "Status Code",
+                        actual = statusCodes[0].attributes.getNamedItem("Value").textContent,
+                        expected = TestCommon.TOP_LEVEL_STATUS_CODES.toString()
+                                .replace("urn:oasis:names:tc:SAML:2.0:status:Success", "")
+                                + TestCommon.SECOND_LEVEL_STATUS_CODES.toString())
+            }
```

Comment:
Separate note, but I'm not sure if it's appropriate to change in this PR:

I found this little tidbit about status codes which makes me think we may want to be able to verify the response has a specific status code. In scenarios where we give the IdP bad data, it sounds like the IdP must respond a certain way. From Core 3.2.1:
```
If a SAML responder deems a request to be invalid according to SAML syntax or processing rules,
then if it responds, it MUST return a SAML response message with a <StatusCode> element
with the value urn:oasis:names:tc:SAML:2.0:status:Requester. In some cases, for example
during a suspected denial-of service attack, not responding at all may be warranted.
```

Reply:
This is the case where we provide it bad data. Should I just check that `urn:oasis:names:tc:SAML:2.0:status:Requester` exists?

Reply:
Yeah maybe we have convenience methods like `verifyRequesterError` and `verifyVersionMismatch` for the two errors that we can actually expect in various scenarios. I don't think we would need a `verifyResponderError` until we do SP tests.